triggers.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "correction.h"
+#include "corrections.h"
+
+namespace correction {
+
+class TrigCorrProvider : public CorrectionsBase<TrigCorrProvider> {
+public:
+    enum class UncSource : int {
+        Central = -1,
+        tautrg_ditau_DM0 = 0,
+        tautrg_ditau_DM1 = 1,
+        tautrg_ditau_3Prong = 2,
+        mutrg_singleMu = 3,
+        eletrg_singleEle = 4,
+    };
+    using wpsMapType = std::map<Channel, std::vector<std::pair<std::string, int> > >;
+    static bool isTwoProngDM(int dm)
+    {
+        static const std::set<int> twoProngDMs = { 5, 6 };
+        return twoProngDMs.count(dm);
+    }
+    static const std::string& getTauScaleStr(UncScale scale)
+    {
+        static const std::map<UncScale, std::string> tau_names = {
+            { UncScale::Down, "down" },
+            { UncScale::Central, "nom" },
+            { UncScale::Up, "up" },
+        };
+        return tau_names.at(scale);
+    }
+    static const std::string& getMuScaleStr(UncScale scale)
+    {
+        static const std::map<UncScale, std::string> tau_names = {
+            { UncScale::Down, "systdown" },
+            { UncScale::Central, "sf" },
+            { UncScale::Up, "systup" },
+        };
+        return tau_names.at(scale);
+    }
+    static bool sourceApplies_tau_fromCorrLib(UncSource source, int decayMode, const std::string& trg_type)
+    {
+        if(trg_type=="ditau"){
+            if(source == UncSource::tautrg_ditau_DM0 && decayMode == 0) return true;
+            if(source == UncSource::tautrg_ditau_DM1 && ( decayMode == 1 || decayMode == 2 )) return true;
+            if(source == UncSource::tautrg_ditau_3Prong && ( decayMode == 10 || decayMode == 11 )) return true;
+        }
+        return false;
+    }
+
+    TrigCorrProvider(const std::string& tauFileName, const std::string& deepTauVersion, const wpsMapType& wps_map, const std::string& muFileName, const std::string& period,  const std::string& eleFileName) :
+        tau_corrections_(CorrectionSet::from_file(tauFileName)),
+        tau_trg_(tau_corrections_->at("tau_trigger")),
+        deepTauVersion_(deepTauVersion),
+        wps_map_(wps_map),
+        mu_corrections_(CorrectionSet::from_file(muFileName)),
+        mu_trg_(mu_corrections_->at("NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight")),
```

Comment:
this is year dependent -> pass as an argument

Reply:
I already pass the period in the evaluate part

Reply:
you need to add map/dict in c++/python that relates era to the entry name.